Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```diff
@@ -40,17 +114,17 @@ function CreateProjectGuidAndPathMap
 function IsComponentFormatValid($tokens)
 {
     #expected format for component '[SonarQube project key]:[SonarQube project value]:[MSBuild project guid]:[file name relative to MSBuild project file path]'
-    $isFormatValid = $false
 
     if (!$tokens)
     {
         Write-Verbose "IsComponentFormatValid: tokens is invalid"
-        return $isFormatValid
+        return $false
```

Comment:
Might want to add TryParseComponent (similar to GetRelativeFilePath lines 155-156) that will return a valid object (with the right types i.e. guid instead of string etc.) as properties.